attestation-service/verifier/src/snp/mod.rs

```diff
@@ -82,6 +84,11 @@ impl Verifier for Snp {
             cert_chain,
         } = serde_json::from_slice(evidence).context("Deserialize Quote failed.")?;
 
+        let cert_chain = match cert_chain {
+            Some(c) => c,
+            None => Vec::new(),
+        };
```

Comment:
I think you should just bail here if we don't have a cert chain. You can use a `let else` like you do in a couple other places.